src/transaction/log_meta.hpp

```diff
@@ -50,6 +57,14 @@ namespace cublog
     private:
       using checkpoint_container_t = std::map<log_lsa, checkpoint_info>;
 
+    private:
+      /* flag parallel to 'log global header is shutdown':
+       *  - false: server has not been clean shut down
+       *  - true: has been clean shut down
+       * after start, flag is set to 'false' and the meta log is saved
+       */
+      bool m_is_tsrs_shutdown = false;
```

Comment:
Naming is not ideal. It is intended to be different than `log_Gl.hdr.is_shutdown`.

Reply:
My suggestion is to remove `_tsrs` from the name. Although it is used in the context of transaction server with remote storage for now, the flag may replace `log_Gl.hdr.is_shutdown` completely. The name of a variable should generally reflect its function, not the context that it is used in.
```suggestion
      bool m_is_shutdown = false;
```